Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h

```diff
@@ -103,9 +103,15 @@ namespace AZ::DocumentPropertyEditor::Nodes
         static constexpr auto OnChanged = CallbackAttributeDefinition<void(const Dom::Value&, ValueChangeType)>("OnChanged");
         static constexpr auto Value = AttributeDefinition<AZ::Dom::Value>("Value");
         static constexpr auto ValueType = TypeIdAttributeDefinition("ValueType");
-        static constexpr auto Disabled = AttributeDefinition<bool>("Disabled");
         static constexpr auto ValueHashed = AttributeDefinition<AZ::Uuid>("ValueHashed");
 
+        //! Use this callback attribute if there is need to enable/disable property editors at runtime.
+        static constexpr auto SetDisabled = CallbackAttributeDefinition<void(bool shouldDisable)>("SetDisabled");
+        static constexpr auto Disabled = AttributeDefinition<bool>("Disabled");
+        //! In some cases, a node may need to know that it is descended from a disabled ancestor. For example, disabled
+        //! elements of a disabled container might require different treatment than disabled elements of an enabled container.
+        static constexpr auto AncestorDisabled = AttributeDefinition<bool>("AncestorDisabled");
```

Comment:
Should this be private? This looks like an internal state rather than an attribute the user would be setting, correct?

Reply:
It's true that we use this attribute internally, but I'm not sure if we want to preclude users from using it if they found a need.

Also, I'm not sure the DPE design has room for private attributes as is. @alexmontAmazon might be able to shed some more light on that or correct me if I'm wrong.

Reply:
the way Nick designed this, the PropertyEditorNodes is only a container for attributes, so they don't really consume them themselves, therefore this must be public. It makes sense to keep the handling of this kind of attribute private to the LegacyReflectionBridge, which is already the case. On a related note, the DPE class itself has waaaay too many public methods just so the DPERowWidget can access them, which is poor design. I'm going to make the row a friend of the DPE and then to do a pass making methods private/protected.